From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Sat Nov 2 06:05:23 2024 +0900
Subject: network: introduce network_config_source_from_string()

It is currently unused, but will be used later.
Preparation for later commits.

Bug-Ubuntu: https://launchpad.net/bugs/2098183
Origin: backport, https://github.com/systemd/systemd/commit/bf6e9b383d6907c8c02fdd493cf529a83cdacd8b
[chengen - adjust the context in networkd-util.h.]

--- systemd-255.4.orig/src/network/networkd-util.c
+++ systemd-255.4/src/network/networkd-util.c
@@ -23,7 +23,7 @@ static const char * const network_config
         [NETWORK_CONFIG_SOURCE_RUNTIME] = "runtime",
 };
 
-DEFINE_STRING_TABLE_LOOKUP_TO_STRING(network_config_source, NetworkConfigSource);
+DEFINE_STRING_TABLE_LOOKUP(network_config_source, NetworkConfigSource);
 
 int network_config_state_to_string_alloc(NetworkConfigState s, char **ret) {
         static const char* states[] = {
--- systemd-255.4.orig/src/network/networkd-util.h
+++ systemd-255.4/src/network/networkd-util.h
@@ -57,6 +57,7 @@ CONFIG_PARSER_PROTOTYPE(config_parse_ip_
 CONFIG_PARSER_PROTOTYPE(config_parse_mud_url);
 
 const char *network_config_source_to_string(NetworkConfigSource s) _const_;
+NetworkConfigSource network_config_source_from_string(const char* s) _pure_;
 
 int network_config_state_to_string_alloc(NetworkConfigState s, char **ret);
 
